### PR TITLE
Switch to Tooltips using floating-ui

### PR DESCRIPTION
**Pull request by cquiroz**
I was never too happy with SUI tooltips as they are kind of ugly and may not place themselves on the right place.

This PR includes a facade to use [floating-ui](https://floating-ui.com/) which is a low-level library to implement thing like tooltips. It doesn't include a Tooltip itsef thus a component is provided.

The tooltip looks better IMHO:
* before
Explore 2022-08-30 16-54-40(1)

* after:
Explore 2022-08-30 16-56-20

A nice feature is that it can adapt its location depending on the size available as can be seen here:

https://user-images.githubusercontent.com/3615303/187576984-eff7ff82-abd5-4d29-98f1-d9bc391a3340.mp4

### CODE REVIEW

File: common/src/main/scala/react/floatingui/Tooltip.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import js.annotation._
+
+/**
+ * Tooltip base on floating ui see: https://floating-ui.com/docs/react-dom
+ */
+case class Tooltip(trigger: VdomTag, tooltip: VdomNode, placement: Placement = Placement.Top)
+    extends ReactFnProps[Tooltip](Tooltip.component)
+
+object Tooltip {
+  type Props = Tooltip
```
Comment (rpiaggio) [inline common/src/main/scala/react/floatingui/Tooltip.scala:32]: Can we make `Props` and `component` private and protected?

Comment (cquiroz) [reply common/src/main/scala/react/floatingui/Tooltip.scala:32]: Are we doing that often?

File: common/src/main/scala/react/floatingui/Tooltip.scala [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import scala.scalajs.js.annotation.JSExportTopLevel
+import scala.scalajs.js.|
+
+import js.annotation._
+
+/**
+ * Tooltip base on floating ui see: https://floating-ui.com/docs/react-dom
+ */
+case class Tooltip(trigger: VdomTag, tooltip: VdomNode, placement: Placement = Placement.Top)
+    extends ReactFnProps[Tooltip](Tooltip.component)
```
Comment (rpiaggio) [inline common/src/main/scala/react/floatingui/Tooltip.scala:29]: With Scala 3 we don't need the `React(Fn)Props` type parameter anymore. We can write eg: `extends ReactFnProps(Tooltip.component)`

File: common/src/main/scala/react/floatingui/Tooltip.scala [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSExportTopLevel
+import scala.scalajs.js.|
+
+import js.annotation._
+
+/**
+ * Tooltip base on floating ui see: https://floating-ui.com/docs/react-dom
+ */
+case class Tooltip(trigger: VdomTag, tooltip: VdomNode, placement: Placement = Placement.Top)
```
Comment (rpiaggio) [inline common/src/main/scala/react/floatingui/Tooltip.scala:28]: Is the idea to move this to `lucuma-react` eventually?

File: common/src/main/scala/react/floatingui/hooks.scala [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.|
+
+@js.native
+trait ElementProps extends js.Object
+
+type ElementPropsList = List[ElementProps]
+
+object HooksApiExt {
+  val floatingHook =
```
Comment (rpiaggio) [inline common/src/main/scala/react/floatingui/hooks.scala:18]: Can we make the hook `val`s private or at least protected?

File: common/src/main/webapp/theme/semantic.less [hunk 5]
```diff
@@ -130,9 +130,9 @@
 & {
   @import "fomantic-ui-less/definitions/modules/nag";
 }
-& {
-  @import "fomantic-ui-less/definitions/modules/popup";
-}
+// & {
```
Comment (rpiaggio) [inline common/src/main/webapp/theme/semantic.less:133]: Commented code

Comment (rpiaggio) [reply common/src/main/webapp/theme/semantic.less:133]: Maybe the idea is to leave it like this?

**Review by rpiaggio — APPROVED** [review]
LGTM

File: common/src/main/scala/react/floatingui/Tooltip.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/react/floatingui/Tooltip.scala:32]: Some of our components have it already, but I think we should.

File: common/src/main/webapp/theme/semantic.less (hunk 5 above)
Comment (cquiroz) [reply common/src/main/webapp/theme/semantic.less:133]: Yes I'm going to leave it there, we could remove a few more and reduce css size

File: common/src/main/scala/react/floatingui/Tooltip.scala (hunk 3 above)
Comment (cquiroz) [reply common/src/main/scala/react/floatingui/Tooltip.scala:28]: Yes we should. I want to keep it here for a while until I'm sure it is working properly. The facade is very minimal and may need to be expanded

Comment (toddburnside) [inline common/src/main/scala/react/floatingui/Tooltip.scala:28]: I wonder if we should have a `disabled` flag or something? I'm thinking of a situation where we have a element that will expect to always be in the DOM, but will only have a tooltip some of the time.

Another situation could be where we want the tooltip to always be displayed. For example, the way we currently display errors for inputs.

**Review by toddburnside — APPROVED** [review]
LGTM with a couple of comments.

primereact has their own tooltip component. It's functionality is built into many of their components which makes it easy to use, but less flexible than this implementation.

File: common/src/main/scala/react/floatingui/Tooltip.scala (hunk 3 above)
Comment (cquiroz) [reply common/src/main/scala/react/floatingui/Tooltip.scala:28]: Right, we could add those as needed

**Pull request merged**